assets/app/scripts/controllers/create.js

```diff
@@ -47,33 +119,198 @@ angular.module('openshiftConsole')
       return false;
     }
 
-    function updateTemplates() {
-      // Check if we've loaded templates from both the openshift and project namespaces.
-      $scope.templatesLoaded = openshiftTemplates && projectTemplates;
-      $scope.instantApps = {};
-      $scope.otherTemplates = {};
-
-      // Categorize templates as instant apps or "other."
-      var categorizeTemplates = function(template) {
-        var uid = uidFilter(template);
-        if (isInstantApp(template)) {
-          $scope.instantApps[uid] = template;
+    function matchesKeyword(name, tags, keyword) {
```

Comment:
so we are only matching keyword against the name or the tags?  did you consider also matching on the description